oshdb-filter/src/main/java/org/heigit/ohsome/oshdb/filter/FilterParser.java

```diff
@@ -206,15 +215,51 @@ public FilterParser(TagTranslator tt, boolean allowContributorFilters) {
         ),
         Scanners.isChar(')')
     );
+    final Parser<ValueRange> positiveIntegerRange = Parsers.between(
+        Scanners.isChar('('),
+        Parsers.or(
+            Parsers.sequence(number, dotdot, number,
+                (min, ignored, max) -> new ValueRange(min, max)),
+            number.followedBy(dotdot).map(
+                min -> new ValueRange(min, Double.POSITIVE_INFINITY)),
+            Parsers.sequence(dotdot, number).map(
+                max -> new ValueRange(0, max))
+        ),
+        Scanners.isChar(')')
+    );
     final Parser<GeometryFilter> geometryFilterArea = Parsers.sequence(
         area, colon, floatingRange
     ).map(GeometryFilterArea::new);
     final Parser<GeometryFilter> geometryFilterLength = Parsers.sequence(
         length, colon, floatingRange
     ).map(GeometryFilterLength::new);
+    final Parser<GeometryFilter> geometryFilterPerimeter = Parsers.sequence(
+        perimeter, colon, floatingRange
+    ).map(GeometryFilterPerimeter::new);
+    final Parser<GeometryFilter> geometryFilterVertices = Parsers.sequence(
+        vertices, colon, positiveIntegerRange
+    ).map(GeometryFilterVertices::new);
+    final Parser<GeometryFilter> geometryFilterOuters = Parsers.sequence(
+        outers, colon, Parsers.or(positiveIntegerRange, number.map(n -> new ValueRange(n, n)))
+    ).map(GeometryFilterOuterRings::new);
+    final Parser<GeometryFilter> geometryFilterInners = Parsers.sequence(
+        inners, colon, Parsers.or(positiveIntegerRange, number.map(n -> new ValueRange(n, n)))
+    ).map(GeometryFilterInnerRings::new);
+    final Parser<GeometryFilter> geometryFilterRoundness = Parsers.sequence(
+        roundness, colon, floatingRange
+    ).map(GeometryFilterRoundness::new);
+    final Parser<GeometryFilter> geometryFilterSquareness = Parsers.sequence(
+        squareness, colon, floatingRange
+    ).map(GeometryFilterSquareness::new);
```

Comment:
I didn't find an easy way to limit input values to given values in the parsing stage. Throwing an exception in the constructor (e.g. of `GeometryFilterSquareness`) would work, but I think there's not a whole lot of benefit to do it IMO. (If needed, this could be implemented at a later stage, though).